src/test/java/com/thirdcc/webapp/web/rest/EventResourceIT.java

```diff
@@ -242,6 +240,54 @@ public void getAllEvents() throws Exception {
             .andExpect(jsonPath("$.[*].status").value(hasItem(DEFAULT_STATUS.toString())));
     }
 
+    @Test
+    @Transactional
+    public void getAllUpcomingEvents() throws Exception {
+        // Initialize the database
+        Event savedEvent = createEntity();
+        savedEvent.setStartDate(Instant.now().plus(20, ChronoUnit.DAYS));
+        eventRepository.saveAndFlush(savedEvent);
+
+        // Get all the eventList
+        restEventMockMvc.perform(get("/api/events?sort=id,desc"))
+            .andExpect(status().isOk())
+            .andExpect(content().contentType(MediaType.APPLICATION_JSON_UTF8_VALUE))
+            .andExpect(jsonPath("$.[*].id").value(hasItem(savedEvent.getId().intValue())))
+            .andExpect(jsonPath("$.[*].name").value(hasItem(DEFAULT_NAME)))
+            .andExpect(jsonPath("$.[*].description").value(hasItem(DEFAULT_DESCRIPTION)))
+            .andExpect(jsonPath("$.[*].remarks").value(hasItem(DEFAULT_REMARKS)))
+            .andExpect(jsonPath("$.[*].venue").value(hasItem(DEFAULT_VENUE)))
+            .andExpect(jsonPath("$.[*].startDate").value(hasItem(savedEvent.getStartDate().toString())))
+            .andExpect(jsonPath("$.[*].endDate").value(hasItem(DEFAULT_END_DATE.toString())))
+            .andExpect(jsonPath("$.[*].fee").value(hasItem(DEFAULT_FEE.intValue())))
+            .andExpect(jsonPath("$.[*].requiredTransport").value(hasItem(DEFAULT_REQUIRED_TRANSPORT)))
+            .andExpect(jsonPath("$.[*].status").value(hasItem(DEFAULT_STATUS.toString())));
+    }
+
+    @Test
+    @Transactional
+    public void getAllPastEvents() throws Exception {
+        // Initialize the database
+        Event savedEvent = createEntity();
+        savedEvent.setStartDate(Instant.now().minus(20, ChronoUnit.DAYS));
+        eventRepository.saveAndFlush(savedEvent);
+
+        // Get all the eventList
+        restEventMockMvc.perform(get("/api/events?sort=id,desc"))
```

Comment:
this API correct?

Reply:
Fixed in 0776434d822e3f44812360d825e6a04ca8933805